addons/crm/models/res_partner.py

```diff
@@ -52,6 +52,19 @@ def _compute_opportunity_count(self):
                     partner.opportunity_count += group['partner_id_count']
                 partner = partner.parent_id
 
+    @api.onchange('parent_id', 'company_type')
+    def _onchange_parent_id_for_team_id(self):
+        # While creating / updating parent, take the parent salesteam value and set on current record
```

Comment:
Could be the docstring instead of a comment.
Also a bit outdated apparently, it's not "while create/updating", it's when changing it on the form view.

Why not something like:
"When changing the parent on a partner, attempt to synchronize the team_id if we're on a person".